exporters/dynamicconfig/confignotifier.go

```diff
@@ -0,0 +1,200 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamicconfig
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	controllerTime "go.opentelemetry.io/otel/sdk/metric/controller/time"
+)
+
+type MetricConfig struct {
+	Period time.Duration
+}
+
+// TODO: Replace with something better when we have a solid config spec
+func (config *MetricConfig) equals(otherConfig *MetricConfig) bool {
+	if config == nil && otherConfig == nil {
+		return true
+	}
+
+	if config == nil || otherConfig == nil {
+		return false
+	}
+
+	configString := fmt.Sprintf("%+v", *config)
+	otherConfigString := fmt.Sprintf("%+v", *otherConfig)
+
+	return configString == otherConfigString
+}
+
+// TODO: Read actual config from host
+// We don't do anything with the configHost string right now
+// TIMES_READ_TEST is for testing purposes, to mimic a dynamic service
+// The second time we call readConfig, the sampling period changes
+var TIMES_CONFIG_READ_TEST = 0
+
+func readConfig(configHost string) *MetricConfig {
+	period := 10 * time.Second
+
+	if TIMES_CONFIG_READ_TEST > 0 {
+		period = 2 * time.Second
+	}
+
+	TIMES_CONFIG_READ_TEST++
+
+	return &MetricConfig{
+		Period: period,
+	}
+}
+
+type ConfigWatcher interface {
+	OnInitialConfig(config *MetricConfig)
+	OnUpdatedConfig(config *MetricConfig)
+}
+
+// A ConfigNotifier monitors a config service for a config changing
+// It then lets all it's subscribers know if the config has changed
+//
+// ch is used to shut down the config checking routine when we stop ConfigNotifier
+// checkFrequency is how often we check to see if the config service has changed
+// clock is added for testing time-related functionality
+// config is the current config
+// configHost is the IP address of the config service. Can be set to ""
+// subscribed is a map of all configNotifier's current subscribers
+// ticker controls when we check the config service for a new config
+// wg is used to wait for the config-checking routine to return before stopping
```

Comment:
Is this the go way of adding comment for struct field?

Reply:
Doing it by comment by comment seems more widely used, so changed it!